core/alsp_src/generic/bsio.c

```diff
@@ -3675,6 +3719,57 @@ ssbq_get_msg()
 }
 #endif /* SSBQ */
 
+
+/*
+ * sio_set_console_prompt(Prompt)
+ */
+
+int
+sio_set_console_prompt()
+{
+    PWord v1;
+    int   t1;
+
+    w_get_An(&v1, &t1, 1);
+
+    if (!getstring((UCHAR **)&console_prompt, v1, t1)){
+	console_prompt = "?- ";
```

Comment:
I think sio_set_console_prompt/1 and sio_set_history_file/1 should fail if their argument isn't an atom (or maybe throw an exception). This will both eliminate the constant strings and communicate the error back to Prolog.